<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Richard Collins &lt;richard.collins@bristol.ac.uk&gt;</author>
 <comment>Copyright (C) University of Bristol 2017
   This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
   If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
   See LICENSE file for details.</comment>
 <class>KeyViewer</class>
 <widget class="QMainWindow" name="KeyViewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>688</width>
    <height>446</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QKD Key Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="sitesTab">
       <attribute name="title">
        <string>Sites</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="lblSiteAgent">
          <property name="text">
           <string>Local Site Agent</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QComboBox" name="localSiteAgents">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="localAgentGo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="go-next">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>RemoteConnection</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QListWidget" name="knownSites">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="getNewKey">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Get A New Key</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="getExistingKey">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Existing Key</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearKey">
            <property name="text">
             <string>Clear</string>
            </property>
            <property name="icon">
             <iconset theme="edit-clear">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Key ID</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="keyId">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Key Value</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QStackedWidget" name="keyStack">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="keyActionsPage">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="sendToHSM">
                <property name="text">
                 <string>Send To HSM</string>
                </property>
                <property name="icon">
                 <iconset theme="mail-forward">
                  <normaloff>.</normaloff>.</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="addModule">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Add PKCS#11 module</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset theme="add">
                  <normaloff>.</normaloff>.</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="revealKey">
              <property name="text">
               <string>Reveal Key</string>
              </property>
              <property name="icon">
               <iconset theme="dialog-warning">
                <normaloff>.</normaloff>.</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="keyValuePage">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPlainTextEdit" name="keyValue">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="hsmTab">
       <attribute name="title">
        <string>HSM</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="3" column="2" colspan="2">
         <widget class="QLineEdit" name="toHSM"/>
        </item>
        <item row="4" column="3">
         <widget class="QPushButton" name="sendToHsm">
          <property name="text">
           <string>Send to HSM</string>
          </property>
          <property name="icon">
           <iconset theme="forward">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="2">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Keys</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>To Device</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QToolButton" name="openDestHsm">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="fileopen">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QSpinBox" name="keysToSend">
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>For destination</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2" colspan="2">
         <widget class="QLineEdit" name="fromHSM"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>From Device</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QToolButton" name="openHsm">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="fileopen">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="7" column="3">
         <widget class="QPushButton" name="clearHSM">
          <property name="text">
           <string>Clear HSM</string>
          </property>
          <property name="icon">
           <iconset theme="edit-clear">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="2" colspan="2">
         <widget class="QComboBox" name="destinationCbo">
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>688</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
